README.md

```diff
@@ -1,58 +1,237 @@
+| Merge | boolean | Boolean value [true/false] - When set to true, will merge all of the remote files into a single PDF file. All of the remote files MUST be images. </br></br>By default this field is set to **false**. |
+| Metadata | string | JSON object or JSON array of objects containing metadata fields for a given remote file. </br></br>When merge field is set to **false**:</br></br>When using the urls field this should be a JSON object array that matches the urls field array length.</br></br>When using the url field the metadata field should be a JSON object.</br></br>When the merge field is set to true: The metadata field should be a JSON object.|
+| PropagateMetadata | boolean | This parameter enables the specification of a single metadata object to be applied across multiple files from remote URLs or remote sources. When used, `propagate_metadata` ensures that the defined metadata is consistently attached to all the specified files during their upload and processing. This feature is particularly useful for maintaining uniform metadata across a batch of files, streamlining data organization and retrieval. |
+| ParentFilePrefix | string | The `parent_file_prefix` parameter is used to specify a virtual folder destination for the uploaded files, diverging from the default 'Inbox' folder. By setting this parameter, users can organize files into specific virtual directories, enhancing file management and accessibility within Alfred's system. |
+
+##### Step by Step Guide
+
+1. The first step after initializing the Alfred client is to create a request object with the necessary parameters. The following parameters are available for the `FileUploadRequest` object:</br></br>
+
+   ```csharp
+   // Create a request object with the necessary parameters
+   UploadRequest uploadRequest = new UploadRequest
+   {
+      Urls = new List<string> { "https://pdfobject.com/pdf/sample.pdf", "https://pdfobject.com/pdf/sample.pdf" },
+      FileNames = new List<string> { "sample.pdf", "sample.pdf" },
+
+   };
+   ```
+
+2. Next, call the appropriate method to upload the file. In this case, we'll use the `UploadAsync` method to upload the file from the URL:</br></br>
 
    ```csharp
    // Upload remote file
-   var uploadResult = await alfred.File.UploadAsync(new FileUploadRequest
+   var uploadResult = await alfred.File.UploadAsync(uploadRequest);
+   ```
+
+3. Handle the response accordingly. The response will contain the job ID since the `UploadAsync` method triggers job processing:</br></br>
+
+   ```csharp
+   Console.WriteLine(uploadResult.JobId);
+   ```
+
+4. Finally you can get the job status using the `Job` domain. The following parameters are available for the `GetAsync` method:</br></br>
+
+   | Parameter | Type | Description |
+   | --- | --- | --- |
+   | JobId | Guid | Job ID |
+
+   **Example:**</br></br>
+
+   ```csharp
+   // Get job status
+   var jobStatus = await alfred.Job.GetAsync(uploadResult.JobId);
```

Comment:
> 4. How do I know its done?

After this, we could add information on job stages, what are the most important stages and which one determines that the process is done.